src/modules/adaptors/Aura/AuraERC4626Adaptor.sol

```diff
@@ -0,0 +1,79 @@
+    // Where:
+    // `auraPool` is the AURA pool address position this adaptor is working with.
+    //================= Configuration Data Specification =================
+    // NA
+    //====================================================================
+
+    /**
+     * @notice Attempted to interact with an auraPool the Cellar is not using.
+     */
+    error AuraExtrasAdaptor__AuraPoolPositionsMustBeTracked(address auraPool);
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Aura ERC4626 Adaptor V 0.1"));
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to get rewards for an AuraPool.
+     * @param _auraPool the specified AuraPool
+     * @param _claimExtras Whether or not to claim extra rewards associated to the AuraPool (outside of rewardToken for AuraPool)
+     */
+    function getRewards(IBaseRewardPool _auraPool, bool _claimExtras) public {
+        _validateAuraPool(address(_auraPool));
+        _getRewards(_auraPool, _claimExtras);
+    }
+
+    /**
+     * @notice Validates that a given auraPool is set up as a position in the calling Cellar.
+     * @dev This function uses `address(this)` as the address of the calling Cellar.
+     */
+    function _validateAuraPool(address _auraPool) internal view {
+        bytes32 positionHash = keccak256(abi.encode(identifier(), false, abi.encode(_auraPool)));
+        uint32 positionId = Cellar(address(this)).registry().getPositionHashToPositionId(positionHash);
+        if (!Cellar(address(this)).isPositionUsed(positionId))
+            revert AuraExtrasAdaptor__AuraPoolPositionsMustBeTracked(_auraPool); // TODO: troubleshoot uncommented implementation code here
+    }
+
+    //============================================ Interface Helper Functions ===========================================
+
+    //============================== Interface Details ==============================
+    // It is unlikely, but AURA pool interfaces can change between versions.
+    // To account for this, internal functions will be used in case it is needed to
+    // implement new functionality.
+    //===============================================================================
+
+    function _getRewards(IBaseRewardPool _auraPool, bool _claimExtras) internal virtual {
```

Comment:
Oh did you separate this into a virtual internal function because they might change the `getReward` function in the future? Kinda like you did with Fraxlend adaptors?

Reply:
Yes, that was the intention behind having a helper function. I did not discuss the % likelihood of new pool versions with the Aura team, but you never know so I thought it was a fair add-on. I can remove it and make it directly part of the `getRewards()` function though. wdyt?